elastic4s-handlers/src/main/scala/com/sksamuel/elastic4s/handlers/fields/DenseVectorFieldBuilderFn.scala

```diff
@@ -17,6 +31,14 @@ object DenseVectorFieldBuilderFn {
     builder.field("dims", field.dims)
     builder.field("index", field.index)
     builder.field("similarity", field.similarity.name)
+    field.indexOptions.foreach { options =>
```

Comment:
I think we need to also check if `index` is `true` here, because "This parameter can only be specified when index is true." according to the docs.

Reply:
add a filter and tests